Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeDelegate.cpp

```diff
@@ -184,6 +184,14 @@ namespace AzToolsFramework
 
         void ComponentModeDelegate::OnComponentModeEnterButtonPressed()
         {
+            // Check the entity hasn't been deselected but we haven't been told yet.
+            bool isSelected = false;
+            ToolsApplicationRequests::Bus::BroadcastResult(isSelected, &ToolsApplicationRequests::IsSelected, m_entityComponentIdPair.GetEntityId());
```

Comment:
One thought I just has is we could use the `IsSelected` helper instead of calling the Bus directly, just makes the code a bit cleaner (the call is in EditorEntityHelpers.h, though it does interestingly actually use a different EBus, so might be worth double checking 🤔)

Reply:
@hultonha Thanks, I've switched to the EditorEntityModel version as it asks the entity directly rather than searching through a list of selected entities. Is there a case for removing the ToolsApplication version? Storing the selected information in two different ways seems like asking for trouble.

Reply:
That's great to hear thanks @sphrose! 👍 If we can remove the one in ToolsApplication I'd be all for it, we just have to be careful if many other callers are using it (we can definitely leave this for a separate PR) as this would count as a deprecation. We might need to follow the guidance here if it is used very widely - https://github.com/o3de/community/blob/main/guides/o3de-deprecation-guidelines.md.